label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleValidatorImpl.java

```diff
@@ -120,7 +135,26 @@ private void validateJsonCondition(JsonCondition jsonCondition) {
   }
 
   private void validateStringCondition(StringCondition stringCondition) {
-    validateNonDefaultPresenceOrThrow(stringCondition, stringCondition.OPERATOR_FIELD_NUMBER);
+    validateNonDefaultPresenceOrThrow(stringCondition, StringCondition.OPERATOR_FIELD_NUMBER);
+    if (stringCondition.getOperator().equals(OPERATOR_MATCHES_IPS)
+        || stringCondition.getOperator().equals(OPERATOR_NOT_MATCHES_IPS)) {
+      if (stringCondition.getKindCase().equals(StringCondition.KindCase.VALUE)) {
+        validateNonDefaultPresenceOrThrow(stringCondition, StringCondition.VALUE_FIELD_NUMBER);
```

Comment:
not needed, the case you checked on line 141 checks that this is set.

Might be a bit more readable as a switch in case other cases are added in future. My personal pref is to always use switches when working with enums to avoid that assumption that it will never be added to (and then static analysis can flag missing cases). Here for example, imagine neither value nor values was set. You'd go into the else case assuming it must be values, and then get a cryptic message that the plural version is not set (rather than that the string condition has neither set, which the default case of the switch would throw).

Anyway, not required to do the switch just something to think about.